app/assets/javascripts/discourse/app/lib/load-script.js

```diff
@@ -102,3 +107,17 @@ export default function loadScript(url, opts) {
     }
   });
 }
+
+export function cacheBuster(url) {
+  if (PUBLIC_JS_VERSIONS) {
+    const pathParts = url.split("/");
+    if (pathParts[1] === "javascripts") {
+      const version = PUBLIC_JS_VERSIONS[pathParts[2]];
+      if (version !== undefined) {
```

Comment:
```suggestion
      if (typeof version !== "undefined") {
```
This is preferred due to the unusual case that undefined can be overwritten 🤪